src/dreyfus/src/clouseau_rpc.erl

```diff
@@ -316,3 +326,46 @@ rpc(Ref, Msg) ->
 
 clouseau() ->
     list_to_atom(config:get("dreyfus", "name", "clouseau@127.0.0.1")).
+
+-type service() :: sup | main | analyzer | cleanup.
+-type liveness_status() :: alive | timeout.
+
+-spec clouseau_services(MajorVsn) -> [service()] when
+    MajorVsn :: binary() | null.
+clouseau_services(_MajorVsn) ->
+    [sup, main, analyzer, cleanup].
+
+-spec is_valid(Service) -> boolean() when
+    Service :: atom().
+is_valid(Service) when is_atom(Service) ->
+    lists:member(Service, clouseau_services(clouseau_major_vsn())).
+
+-spec check_service(Service) -> {service(), liveness_status()} | no_return() when
```

Comment:
```erlang
| throw({not_found, binary()})
```

BTW earlier the error in throw was string, but here it is binary.

Reply:
Updated all throwing functions to `throw({atom(), binary()})`.